### PR TITLE
Add Icon to Windows binary

### PR DESCRIPTION
**Pull request by nozwock**
Addressing #9 by implementing solution `#1`

### CODE REVIEW

**Comment by nozwock** [comment]
Currently using `icon-64x64.ico` for Icon resource.

**Comment by nozwock** [comment]
Should I merge these [refactors](https://github.com/nozwock/rusty-autoclicker/commit/242f11185fd7961a943c8a08bb44ca3702f5bd6f) or create a separate PR for it instead?

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// https://github.com/mxre/winres/issues/40
+
+fn main() -> io::Result<()> {
+    if std::env::var("CARGO_CFG_TARGET_FAMILY").unwrap() == "windows" {
+        let mut res = WindowsResource::new();
+        let env = std::env::var("CARGO_CFG_TARGET_ENV").unwrap();
+        match env.as_str() {
+            "gnu" => {
+                // Not sure whether this works the same if build on windows; needs testing
+                // Perfectly fine if cross compiling from linux
```
Comment (MrTanoshii) [inline build.rs:15]: explorer_2omhbhc2SJ

Tested on Windows, it works!

File: Cargo.toml [hunk 2]
```diff
@@ -9,24 +9,28 @@ edition = "2021"
...
 default = []
-persistence = ["eframe/persistence", "serde"] # Enable if you want to persist app state on shutdown
+persistence = [
+    "eframe/persistence",
+    "serde",
+] # Enable if you want to persist app state on shutdown
 
 [profile.release]
-opt-level = 3 # all optimizations
+opt-level = 3     # all optimizations
```
Comment (MrTanoshii) [inline Cargo.toml:33]: Wonky whitespace

File: Cargo.toml [hunk 3]
```diff
@@ -9,24 +9,28 @@ edition = "2021"
...
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
-[lib]
-crate-type = ["cdylib", "rlib"]
-
 [dependencies]
 rand = "0.8"
 device_query = "1.1"
-eframe = "0.19" # Gives us egui, epi and web+native backends
+eframe = "0.19"                                                     # Gives us egui, epi and web+native backends
```
Comment (MrTanoshii) [inline Cargo.toml:15]: Wonky whitespace

**Review by MrTanoshii — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -9,24 +9,28 @@ edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
-[lib]
-crate-type = ["cdylib", "rlib"]
-
```
[the remark this answers is not in the document]
Comment (nozwock) [reply Cargo.toml:14]: You don't need a `lib.rs` for that though.

[the remark this answers is not in the document]
Comment (nozwock) [reply Cargo.toml:14]: And again as I mentioned on https://github.com/MrTanoshii/rusty-autoclicker/issues/9#issuecomment-1305145691, `winres` wouldn't work with pkg consisting of both a lib and main crate.

File: Cargo.toml (hunk 3 above)
Comment (nozwock) [reply Cargo.toml:15]: ya I don't know, that's probably some toml formatter I have.

File: build.rs (hunk 1 above)
Comment (nozwock) [reply build.rs:15]: Great!

File: build.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                res.set_ar_path("x86_64-w64-mingw32-ar")
+                    .set_windres_path("x86_64-w64-mingw32-windres");
+            }
+            "msvc" => {}
+            _ => panic!("unsupported target-env: {}", env),
+        };
+        res.set_icon("assets/icon-64x64.ico");
```
[the remark this answers is not in the document]
Comment (nozwock) [reply build.rs:22]: It's fine, I already made the `.ico` file.

File: Cargo.toml (hunk 2 above)
Comment (nozwock) [reply Cargo.toml]: Fixed

**Comment by MrTanoshii** [comment]
> Btw should I merge these few [refactors](https://github.com/nozwock/rusty-autoclicker/commit/242f11185fd7961a943c8a08bb44ca3702f5bd6f) aswell or create a separate PR for it instead?

Separate PR please

File: Cargo.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (MrTanoshii) [reply Cargo.toml:14]: Sure, let's get rid of lib and we'll deal with the structure at a later time

File: src/lib.rs [hunk 6]
```diff
@@ -1,8 +0,0 @@
-#![forbid(unsafe_code)]
-#![cfg_attr(not(debug_assertions), deny(warnings))] // Forbid warnings in release builds
-#![warn(clippy::all, rust_2018_idioms)]
-
-mod app;
-mod constants;
-
-pub use app::RustyAutoClickerApp;
```
[the remark this answers is not in the document]
Comment (MrTanoshii) [reply src/lib.rs:8]: Disregard

File: Cargo.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (nozwock) [reply Cargo.toml:14]: Got it!

File: build.rs (hunk 1 above)
Comment (nozwock) [reply build.rs:15]: Let's leave this for now, you probably build it for `msvc` since that's the default on Windows.

**Review by nozwock — COMMENTED** [review]
Appropriate changes have been made, hmm I don't understand how reviews are supposed to work...what now :-P

File: build.rs (hunk 1 above)
Comment (MrTanoshii) [reply build.rs:15]: You are right, my bad, I was using the wrong target
This fails on my end

Screenshot 2022-11-07 162020

Comment (nozwock) [reply build.rs:15]: > You are right, my bad, I was using the wrong target This fails on my end

I'm confused, can I know which target you used here?

**Comment by nozwock** [comment]
@MrTanoshii Tests fail because lib doesn't exist anymore.

**Review by MrTanoshii — APPROVED** [review]

**Comment by MrTanoshii** [comment]
> @MrTanoshii Test fails because lib doesn't exist anymore, here
> 
> https://github.com/MrTanoshii/rusty-autoclicker/blob/45961c9ead705fb5d367171d06a746f55e7cbb28/.github/workflows/rust_check.yml#L92
> 
> Should I make changes to the Rust check workflow?

Sure, we can actually get rid of the whole job in a new PR.
https://github.com/MrTanoshii/rusty-autoclicker/blob/45961c9ead705fb5d367171d06a746f55e7cbb28/.github/workflows/rust_check.yml#L73-L92

**Pull request merged**